### ISSUE TITLE
Add support of autodiff wrt trajectory

### ISSUE THREAD

**Issue by chaithyagr** [mind-inria/mri-nufft#103]
The main idea is to implement the differentation of NUFFT forward and adjoint operator with respect to k-space trajectory locations.

Some earlier implementations:

1) TorchKBNUFFT: https://github.com/mmuckley/torchkbnufft/blob/main/torchkbnufft/_autograd/interp.py
2) TFKBNUFFT: https://github.com/zaccharieramzi/tfkbnufft/blob/master/tfkbnufft/kbnufft.py
3) tensorflow-nufft: https://github.com/mrphys/tensorflow-nufft/blob/7d2f502c6f5f4ba305e7a389ef8794b65a5b4f29/tensorflow_nufft/python/ops/nufft_ops.py#L126-L232

1) and 2) are very slow and inefficient. 3) while is much faster, it is based on Google_CUDA and reimplementation of cufinufft.

When implemented in mri-nufft, we expect a much better interface with all the backends, giving a much faster, efficient and flexible way to compute. 

Underlying math: https://web.eecs.umich.edu/~fessler/papers/files/jour/23/wang-23-eao.pdf

**Comment by chaithyagr** [comment]
@alineyyy , you can possibly build on the tests in https://github.com/mind-inria/mri-nufft/blob/master/tests/test_autodiff.py

These tests are an extension of tests in tfkbnufft here: https://github.com/zaccharieramzi/tfkbnufft/blob/master/tfkbnufft/tests/ndft_test.py

All you need is to compare the gradients from NDFT with that of NUFFT that you implement and if they match, you are mathematically good to go and use them in networks.

**Comment by paquiteau** [comment]
Let's also mention 

https://github.com/flatironinstitute/pytorch-finufft/blob/main/pytorch_finufft/functional.py 

Which uses the maths of the paper (and describes the "coordinates ramps")
